src/main/java/com/almostreliable/merequester/wireless/WRScreen.java

```diff
@@ -0,0 +1,37 @@
+package com.almostreliable.merequester.wireless;
+
+import appeng.client.gui.style.ScreenStyle;
+import appeng.client.gui.widgets.BackgroundPanel;
+import appeng.client.gui.widgets.ToolboxPanel;
+import appeng.client.gui.widgets.UpgradesPanel;
+import appeng.menu.SlotSemantics;
+
+import com.almostreliable.merequester.client.RequesterTerminalScreen;
+
+import de.mari_023.ae2wtlib.api.terminal.IUniversalTerminalCapable;
+import de.mari_023.ae2wtlib.api.terminal.WTMenuHost;
+
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.player.Inventory;
+
+public class WRScreen extends RequesterTerminalScreen<WRMenu> implements IUniversalTerminalCapable {
+
+    public WRScreen(WRMenu menu, Inventory playerInventory, Component name, ScreenStyle style) {
+        super(menu, playerInventory, name, style);
+        if (getMenu().isWUT())
+            addToLeftToolbar(cycleTerminalButton());
+
+        widgets.add("upgrades", new UpgradesPanel(getMenu().getSlots(SlotSemantics.UPGRADE), getMenu().getHost()));
+        if (getMenu().getToolbox().isPresent())
+            widgets.add("toolbox", new ToolboxPanel(style, getMenu().getToolbox().getName()));
+        //widgets.add("singularityBackground", new BackgroundPanel(style.getImage("singularityBackground")));
+    }
+
+    @Override
+    public WTMenuHost getHost() {
+        return (WTMenuHost) getMenu().getHost();
+    }
+
+    @Override
+    public void storeState() {}
```

Comment:
```suggestion
    public void storeState() {}

    @Override
    public boolean keyPressed(int keyCode, int scanCode, int keyPressed) {
        if (!super.keyPressed(keyCode, scanCode, keyPressed))
            return checkForTerminalKeys(keyCode, scanCode);
        return true;
    }
```

this will close the terminal if you press the keybind while in the menu, or open other terminals if you press their keybind. (unless something else already handled that press, e.g. because the search bar is focused)